File: 2391. Minimum Amount of Time to Collect Garbage/Minimum Amount of Time to Collect Garbage.java
class Solution {
    public int garbageCollection(String[] garbage, int[] travel) {
        int[] count = new int[3];
        int result = 0;
        int cost = 0;
        for (int i = 0; i < travel.length; i++){
            result = result + garbage[i].length();
            cost = cost + travel[i];
            if(garbage[i+1].contains("M")){
                count[0] = cost;
            }
            if(garbage[i+1].contains("P")){
                count[1] = cost;
            }
            if(garbage[i+1].contains("G")){
                count[2] = cost;
            }
        }
        result = result + garbage[garbage.length-1].length() + count[0] + count[1] + count[2];
        return result;
    }
}

File: 2391. Minimum Amount of Time to Collect Garbage/Minimum Amount of Time to Collect Garbage.py
class Solution:
    def garbageCollection(self, garbage: List[str], travel: List[int]) -> int:
        count = [0,0,0]
        result = 0
        cost = 0
        for i in range(len(travel)):
            result = result + len(garbage[i])
            cost = cost + travel[i]
            if 'M' in garbage[i+1]:
                count[0] = cost
            if 'P' in garbage[i+1]:
                count[1] = cost
            if 'G' in garbage[i+1]:
                count[2] = cost
        result = result + len(garbage[-1]) + sum(count)
        return result
